src/Microsoft.VisualStudio.Web.CodeGeneration.Utils/ProjectContextExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.Extensions.ProjectModel;
+using Microsoft.Extensions.ProjectModel.Resolution;
+using System.Collections.Generic;
+
+namespace Microsoft.VisualStudio.Web.CodeGeneration.Utils
+{
+    public static class ProjectContextExtensions
+    {
+        public static DependencyDescription GetPackage(this IProjectContext context, string name)
+        {
+            Requires.NotNullOrEmpty(name, nameof(name));
+            Requires.NotNull(context, nameof(context));
+
+            return context.PackageDependencies.FirstOrDefault(package => package.Name.Equals(name, StringComparison.OrdinalIgnoreCase));
+        }
+
+        public static IEnumerable<DependencyDescription> GetReferencingPackages(this IProjectContext context, string name)
+        {
+            Requires.NotNullOrEmpty(name, nameof(name));
+            Requires.NotNull(context, nameof(context));
+
+            return context
```

Comment:
This gets called once during the entire scaffolding operation. 
This is actually for an extensibility point which allows users implement custom scaffolders by referencing `Microsoft.VisualStudio.Web.CodeGeneration` package.